test/conversion_test.go

```diff
@@ -0,0 +1,1031 @@
+	}
+
+	if d := cmp.Diff(v1beta1TaskRunExpected, v1beta1TaskRunGot, filterTypeMeta, filterObjectMeta, filterLabels, filterV1beta1TaskRunStatus, filterCondition, filterReleaseAnnotation, filterContainerStateTerminated, filterV1beta1StepState); d != "" {
+		t.Fatalf("-want, +got: %v", d)
+	}
+}
+
+// TestPipelineCRDConversion first creates a v1beta1 Pipeline CRD using v1beta1Clients and
+// requests it by v1Clients to compare with v1 if the conversion has been
+// correctly executed by the webhook. And then it creates the v1 Pipeline CRD using v1Clients
+// and requests it by v1beta1Clients to compare with v1beta1.
+func TestPipelineCRDConversion(t *testing.T) {
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	t.Parallel()
+
+	c, namespace := setup(ctx, t)
+	knativetest.CleanupOnInterrupt(func() { tearDown(ctx, t, c, namespace) }, t.Logf)
+	defer tearDown(ctx, t, c, namespace)
+
+	v1beta1PipelineName := helpers.ObjectNameForTest(t)
+	v1beta1Pipeline := parse.MustParseV1beta1Pipeline(t, fmt.Sprintf(v1beta1PipelineYaml, v1beta1PipelineName, namespace))
+	v1PipelineExpected := parse.MustParseV1Pipeline(t, fmt.Sprintf(v1PipelineYaml, v1beta1PipelineName, namespace))
+
+	if _, err := c.V1beta1PipelineClient.Create(ctx, v1beta1Pipeline, metav1.CreateOptions{}); err != nil {
+		t.Fatalf("Failed to create v1beta1 Pipeline: %s", err)
+	}
+
+	v1PipelineGot, err := c.V1PipelineClient.Get(ctx, v1beta1PipelineName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Couldn't get expected v1 Pipeline for %s: %s", v1beta1PipelineName, err)
+	}
+
+	if d := cmp.Diff(v1PipelineGot, v1PipelineExpected, filterTypeMeta, filterObjectMeta); d != "" {
+		t.Fatalf("-want, +got: %v", d)
+	}
+
+	v1PipelineName := helpers.ObjectNameForTest(t)
+	v1Pipeline := parse.MustParseV1Pipeline(t, fmt.Sprintf(v1PipelineYaml, v1PipelineName, namespace))
+	v1beta1PipelineExpected := parse.MustParseV1beta1Pipeline(t, fmt.Sprintf(v1beta1PipelineYaml, v1PipelineName, namespace))
+
+	if _, err := c.V1PipelineClient.Create(ctx, v1Pipeline, metav1.CreateOptions{}); err != nil {
+		t.Fatalf("Failed to create v1 Pipeline: %s", err)
+	}
+
+	v1beta1PipelineGot, err := c.V1beta1PipelineClient.Get(ctx, v1PipelineName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Couldn't get expected v1beta1 Pipeline for %s: %s", v1PipelineName, err)
+	}
+
+	if d := cmp.Diff(v1beta1PipelineExpected, v1beta1PipelineGot, filterTypeMeta, filterObjectMeta); d != "" {
+		t.Fatalf("-want, +got: %v", d)
+	}
+}
+
+// TestPipelineRunCRDConversion first creates a v1beta1 PipelineRun CRD using v1beta1Clients and
+// requests it by v1Clients to compare with v1 if the conversion has been
+// correctly executed by the webhook. And then it creates the v1 PipelineRun CRD using v1Clients
+// and requests it by v1beta1Clients to compare with v1beta1.
+func TestPipelineRunCRDConversion(t *testing.T) {
+	ctx := withFullEmbeddedStatus(context.Background())
```

Comment:
alpha-test failing due to `embedded-status` set as `minimal`
Currently investigating why the withFullEmbeddedStatus here does not take effect.
https://prow.tekton.dev/view/gs/tekton-prow/pr-logs/pull/tektoncd_pipeline/5579/pull-tekton-pipeline-alpha-integration-tests/1600939404950507520